### PR TITLE
Reduce bundle size: remove BaseComponent from more components

### PR DESCRIPTION
**Pull request by dzearing**
This change removes more BaseComponent usage from:

* Check
* Checkbox
* ChoiceGroup
* ChoiceGroupOption
* Dialog
* TextField
* MaskedTextField
* Customizer

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8476)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx [hunk 1]
```diff
@@ -28,9 +28,13 @@ export class CheckboxBase extends BaseComponent<ICheckboxProps, ICheckboxState>
   constructor(props: ICheckboxProps, context?: any) {
     super(props, context);
 
-    this._warnMutuallyExclusive({
-      checked: 'defaultChecked'
-    });
+    initializeComponentRef(this);
+
+    if (typeof process !== 'undefined' && process.env.NODE_ENV !== 'production') {
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx]: Orthogonal to this PR, but it feels like this check should be moved inside of `warnMutuallyExclusive` eventually

Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx]: I believe it is already, but... a higher level here means no empty function called or bundle impact.

Comment (KevinTCoughlin) [reply packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx]: Makes even more sense!

**Comment by micahgodbolt** [comment]
Was there discussion on how we're supposed to support focusRect for components not based off of BaseComponent anymore?

https://github.com/OfficeDev/office-ui-fabric-react/blob/61ffe7ba8a1f40a61a8335ac9ef9fb6e4dc30fd7/packages/utilities/src/BaseComponent.ts#L52

It seems, unless there is another component on the page that still extends BaseComponent on the page (like Button) that the focus rects aren't working.

**Comment by ecraig12345** [comment]
Having focus rectangles depend on some random initialization logic in BaseComponent is really not a good design. One workaround would be calling initializeFocusRects manually in each converted component that needs it.

**Comment by dzearing** [comment]
I think perhaps a better solution would be to have "initializeFocusRects" called within the top level import files. That way its called just once on component load. We should mark the file as a side effect so that importing `{ Button }` continues to execute the function.

**Pull request merged**